include/matx/generators/random.h

```diff
@@ -186,233 +206,62 @@ template <typename T> class [[deprecated("Use random() operator instead of rando
-   *
-   * @tparam I Unused
-   * @tparam Is Index types
-   * @return Value at index
-   */
-  template <int I = 0, typename ...Is, std::enable_if_t<I == sizeof...(Is), bool> = true>
-  constexpr __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t GetValC(const cuda::std::tuple<Is...>) const {
-    return 0;
-  }
-
-  template <int I = 0, typename ...Is, std::enable_if_t<I < sizeof...(Is), bool> = true>
-  __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t GetValC(const cuda::std::tuple<Is...> tup) const {
-    return GetValC<I+1, Is...>(tup) + cuda::std::get<I>(tup)*Stride(I);
-  }
+  namespace detail {
 
-  /**
-   * Retrieve a value from a random view
-   *
-   * @tparam Is Index type
-   * @param indices Index values
-   */
-  template <typename... Is>
-  inline __MATX_DEVICE__ T operator()(Is... indices) const
-  {
-    T val;
-    if constexpr (RANK == 0) {
-      get_random(val, &states_[0], dist_);
-    }
-    else {
-      get_random(val, &states_[GetValC<0, Is...>(cuda::std::make_tuple(indices...))], dist_);
-    }
-
-    return alpha_ * val + beta_;
+  template< typename T >
+  struct randIntParams{
+    T min_;
+    T max_;
   };
-
-  /**
-   * Get rank of random view
-   *
-   * @return Rank of view
-   */
-  static inline constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank() { return RANK; }
-
-  /**
-   *  Get size of a dimension
-   *
-   * @param dim Dimension to retrieve
-   * @return Size of dimension
-   */
-  constexpr index_t inline __MATX_HOST__ __MATX_DEVICE__ Size(int dim) const
-  {
-    return shape_[dim];
-  }
-
-  /**
-   * Get stride of a dimension
-   *
-   * @param dim Dimension to retrieve
-   * @return Stride of dimension
-   */
-  index_t inline __MATX_HOST__ __MATX_DEVICE__ Stride(int dim) const
-  {
-    return strides_[dim];
-  }
-
-  template <typename ShapeType, typename Executor>
-  __MATX_INLINE__ void PreRun([[maybe_unused]] ShapeType &&shape, [[maybe_unused]] Executor &&ex) noexcept
-  {
-  }
-
-  template <typename ShapeType, typename Executor>
-  __MATX_INLINE__ void PostRun([[maybe_unused]] ShapeType &&shape, [[maybe_unused]] Executor &&ex) noexcept  
-  {
-  }  
-};
-
-  namespace detail {
+  
+  template< typename T >
+  struct randFloatParams{
+    Distribution_t dist_;
+    T alpha_;
+    T beta_;
+  };  
+  
     template <typename T, typename ShapeType>
     class RandomOp : public BaseOp<RandomOp<T, ShapeType>> {
       private:
         using inner_t = typename inner_op_type_t<T>::type;
         static constexpr int RANK = cuda::std::tuple_size<ShapeType>{};
-        Distribution_t dist_;
         cuda::std::array<index_t, RANK> shape_;
         cuda::std::array<index_t, RANK> strides_;
         index_t total_size_;
         curandStatePhilox4_32_10_t *states_;
-        uint64_t seed_;
-        inner_t alpha_;
-        inner_t beta_;
+        uint64_t seed_;     
         bool init_ = false;
         bool device_;
-
+        
+        union{
+        randFloatParams<inner_t> fParams_;
```

Comment:
Looks like the spacing is off here

Reply:
fixed, in the way I think you mean. I indented the values inside the union (like it was a scope), and left the union at the level of the parent class as it was. 

the only other union I could find in our code was at a namespace scope, so I mirrored that spacing here. let me know if you want it a different way.